packages/portal-proto/src/features/GenomicTables/SomaticMutationsTable/SMTableContainer.tsx

```diff
@@ -145,27 +148,55 @@ export const SMTableContainer: React.FC<SMTableContainerProps> = ({
         geneSymbol: geneSymbol,
         genomicFilters: genomicFilters,
         cohortFilters: cohortFilters,
-        caseFilter: { mode: "", root: {} } as FilterSet,
+        caseFilter: caseFilter,
       });
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [searchTermsForGene, genomicFilters, cohortFilters]);
+  }, [searchTermsForGene, genomicFilters, cohortFilters, caseFilter]);
 
-  useEffect(() => {
+  useDeepCompareEffect(() => {
     if (topSSM) {
       const { ssm_id, consequence_type, aa_change = "" } = topSSM;
-      handleSurvivalPlotToggled(
-        ssm_id,
-        consequence_type
+      const description = consequence_type
+        ? `${searchTermsForGene?.geneSymbol ?? ""} ${aa_change} ${humanify({
+            term: consequence_type.replace("_variant", "").replace("_", " "),
+          })}`
+        : "";
+      handleSurvivalPlotToggled(ssm_id, description, "gene.ssm.ssm_id");
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [topSSM, searchTermsForGene?.geneSymbol]);
+
+  useDeepCompareEffect(
+    () => {
+      const shouldHandleSurvivalPlot =
+        topSSM &&
+        !isEqual(selectedSurvivalPlot, previousSelectedSurvivalPlot) &&
+        isEqual(searchTerm, previousSearchTerm) &&
+        !data?.ssms
+          .map(({ ssm_id }) => ssm_id)
+          .includes(selectedSurvivalPlot.symbol);
+      if (shouldHandleSurvivalPlot) {
+        const { ssm_id, consequence_type, aa_change = "" } = topSSM;
+        const description = consequence_type
           ? `${searchTermsForGene?.geneSymbol ?? ""} ${aa_change} ${humanify({
               term: consequence_type.replace("_variant", "").replace("_", " "),
             })}`
-          : "",
-        "gene.ssm.ssm_id",
-      );
-    }
+          : "";
+        handleSurvivalPlotToggled(ssm_id, description, "gene.ssm.ssm_id");
+      }
+    },
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
here too

Reply:
Was it already re-rendering indefinitely before my suggested changes?

Reply:
i didn't notice it after adding useDeepCompare, but it might've been because of handleSurvivalPlotToggled being included in deps array.

Reply:
Yes, that was to consolidate the condition of other useEffect you have on Line 157.

So what I was trying to was combine two useEffect's logic in one as they had the same bit of code. They are both calling `handleSurvivalPlotToggled` based on two different conditionals, right?

Reply:
1 effect has the selected survival plot as a dependency while the other does not. the combined effect updates the selected plot either way so thats why its looping i think